src/controls/default_controls.ts

```diff
@@ -1,117 +1,122 @@
     actionHandler: scalingEqually,
-  });
+  }),
 
-  objectControls.br = new fabric.Control({
+  br: new Control({
     x: 0.5,
     y: 0.5,
-    cursorStyleHandler: scaleStyleHandler,
+    cursorStyleHandler: scaleCursorStyleHandler,
     actionHandler: scalingEqually,
-  });
+  }),
 
-  objectControls.mtr = new fabric.Control({
+  mtr: new Control({
     x: 0,
     y: -0.5,
-    actionHandler: controlsUtils.rotationWithSnapping,
-    cursorStyleHandler: controlsUtils.rotationStyleHandler,
+    actionHandler: rotationWithSnapping,
+    cursorStyleHandler: rotationStyleHandler,
     offsetY: -40,
     withConnection: true,
     actionName: 'rotate',
-  });
+  }),
+};
 
-  if (fabric.Textbox) {
-    // this is breaking the prototype inheritance, no time / ideas to fix it.
-    // is important to document that if you want to have all objects to have a
-    // specific custom control, you have to add it to Object prototype and to Textbox
-    // prototype. The controls are shared as references. So changes to control `tr`
-    // can still apply to all objects if needed.
-    var textBoxControls = (fabric.Textbox.prototype.controls = {});
+export const textboxDefaultControls = {
+  ...defaultControls,
+  mr: new Control({
+    x: 0.5,
+    y: 0,
+    actionHandler: changeWidth,
+    cursorStyleHandler: scaleSkewCursorStyleHandler,
+    actionName: 'resizing',
+  }),
 
-    textBoxControls.mtr = objectControls.mtr;
-    textBoxControls.tr = objectControls.tr;
-    textBoxControls.br = objectControls.br;
-    textBoxControls.tl = objectControls.tl;
-    textBoxControls.bl = objectControls.bl;
-    textBoxControls.mt = objectControls.mt;
-    textBoxControls.mb = objectControls.mb;
+  ml: new Control({
+    x: -0.5,
+    y: 0,
+    actionHandler: changeWidth,
+    cursorStyleHandler: scaleSkewCursorStyleHandler,
+    actionName: 'resizing',
+  }),
+};
 
-    textBoxControls.mr = new fabric.Control({
-      x: 0.5,
-      y: 0,
-      actionHandler: controlsUtils.changeWidth,
-      cursorStyleHandler: scaleSkewStyleHandler,
-      actionName: 'resizing',
-    });
+FabricObject.prototype.controls = {
+  ...(FabricObject.prototype.controls || {}),
+  ...defaultControls,
+};
 
-    textBoxControls.ml = new fabric.Control({
-      x: -0.5,
-      y: 0,
-      actionHandler: controlsUtils.changeWidth,
-      cursorStyleHandler: scaleSkewStyleHandler,
-      actionName: 'resizing',
-    });
-  }
-})(typeof exports !== 'undefined' ? exports : window);
+// this is breaking the prototype inheritance, no time / ideas to fix it.
+// is important to document that if you want to have all objects to have a
+// specific custom control, you have to add it to Object prototype and to Textbox
+// prototype. The controls are shared as references. So changes to control `tr`
+// can still apply to all objects if needed.
+fabric.Textbox.prototype.controls = {
+  ...(fabric.Textbox.prototype.controls || {}),
+  ...textboxDefaultControls,
+};
```

Comment:
This isn't anymore under `if (fabric.Textbox)` meaning that non importing the textbox will create an error.
Let's keep it under if until we can move this in the Textbox file directly.